### PR TITLE
added extra test cases for inference service

### PR DESCRIPTION
**Pull request by pnaik1**
Closes: https://issues.redhat.com/browse/RHOAIENG-2149

## Description
added extra test cases for frontend/src/api/k8s/inferenceServices.ts

## How Has This Been Tested?
npm run test

## Test Impact
 test coverage:
Before:
Beforeinference

After:
AfterInference

## Request review criteria:

Self checklist (all need to be checked):
- [x] The developer has manually tested the changes and verified that the changes work
- [x] Commits have been squashed into descriptive, self-contained units of work (e.g. 'WIP' and 'Implements feedback' style messages have been removed)
- [x] Testing instructions have been added in the PR body (for PRs involving changes that are not immediately obvious).
- [x] The developer has added tests or explained why testing cannot be added (unit tests & storybook for related changes)

If you have UI changes: 

- [ ] Included any necessary screenshots or gifs if it was a UI change.
- [ ] Included tags to the UX team if it was a UI/UX change (find relevant UX in the [SMEs](https://github.com/opendatahub-io/odh-dashboard/tree/main/docs/smes.md) section).

After the PR is posted & before it merges:
- [ ] The developer has tested their solution on a cluster by using the image produced by the PR to `main`

### CODE REVIEW

File: frontend/src/api/k8s/__tests__/inferenceServices.spec.ts [hunk 1]
```diff
@@ -1,26 +1,65 @@
...
   it('should have the right annotations when creating for Kserve', async () => {
     const inferenceService = assembleInferenceService(mockInferenceServiceModalData({}));
 
     expect(inferenceService.metadata.annotations).toBeDefined();
-    expect(inferenceService.metadata.annotations?.['serving.kserve.io/deploymentMode']).toBe(
-      undefined,
-    );
+    expect(
+      inferenceService.metadata.annotations?.['serving.kserve.io/deploymentMode'],
+    ).toStrictEqual(undefined);
```
Comment (christianvogt) [inline frontend/src/api/k8s/__tests__/inferenceServices.spec.ts:53]: Using `toBe` is correct for primitive type assertions or any other time identity equality is expected.
Can also use `toBeUndefined()`.

File: frontend/src/api/k8s/__tests__/inferenceServices.spec.ts [hunk 2]
```diff
@@ -155,3 +219,256 @@ describe('assembleInferenceService', () => {
...
+        apiGroup: 'serving.kserve.io',
+        apiVersion: 'v1beta1',
+        kind: 'InferenceService',
+        plural: 'inferenceservices',
+      },
+      queryOptions: {},
+    });
+  });
+});
+describe('listScopedInferenceService', () => {
```
Comment (christianvogt) [inline frontend/src/api/k8s/__tests__/inferenceServices.spec.ts:256]: `listScopedInferenceService` accepts an optional `labelSelector` param that requires a test case.

File: frontend/src/api/k8s/__tests__/inferenceServices.spec.ts [hunk 3]
```diff
@@ -155,3 +219,256 @@ describe('assembleInferenceService', () => {
...
+    const inferenceServiceMock = mockInferenceServiceK8sResource({});
+    const { name } = inferenceServiceMock.metadata;
+    k8sListResourceMock.mockResolvedValue(mockK8sResourceList([inferenceServiceMock]));
+    const result = await listScopedInferenceService();
+    expect(result).toStrictEqual([inferenceServiceMock]);
+    expect(k8sListResourceMock).toHaveBeenCalledWith({
+      model: InferenceServiceModel,
+      queryOptions: { ns: name },
+    });
+    expect(k8sListResourceMock).toHaveBeenCalledTimes(2);
```
Comment (christianvogt) [inline frontend/src/api/k8s/__tests__/inferenceServices.spec.ts:267]: Should use `toHaveBeenNthCalledWith` if more than 1 call has been made.

File: frontend/src/api/k8s/__tests__/inferenceServices.spec.ts [hunk 4]
```diff
@@ -155,3 +219,256 @@ describe('assembleInferenceService', () => {
...
+      },
+      queryOptions: {},
+    });
+  });
+});
+describe('listScopedInferenceService', () => {
+  it('should return list of scoped inference service', async () => {
+    const inferenceServiceMock = mockInferenceServiceK8sResource({});
+    const { name } = inferenceServiceMock.metadata;
+    k8sListResourceMock.mockResolvedValue(mockK8sResourceList([inferenceServiceMock]));
```
Comment (christianvogt) [inline frontend/src/api/k8s/__tests__/inferenceServices.spec.ts:260]: Should be mocking both projects and inference service.

File: frontend/src/api/k8s/__tests__/inferenceServices.spec.ts [hunk 5]
```diff
@@ -155,3 +219,256 @@ describe('assembleInferenceService', () => {
...
+    const result = await listScopedInferenceService();
+    expect(result).toStrictEqual([inferenceServiceMock]);
+    expect(k8sListResourceMock).toHaveBeenCalledWith({
```
Comment (christianvogt) [inline frontend/src/api/k8s/__tests__/inferenceServices.spec.ts:263]: If we're expecting two calls, there should be two mocks. Projects and inference services.

**Review by dpanshug — APPROVED** [review]
/lgtm

File: frontend/src/api/k8s/__tests__/inferenceServices.spec.ts [hunk 6]
```diff
@@ -155,3 +207,309 @@ describe('assembleInferenceService', () => {
...
+    const result = await listInferenceService();
+    expect(result).toStrictEqual([inferenceServiceMock]);
+    expect(k8sListResourceMock).toHaveBeenCalledTimes(1);
+    expect(k8sListResourceMock).toHaveBeenCalledWith({
+      model: {
+        apiGroup: 'serving.kserve.io',
+        apiVersion: 'v1beta1',
+        kind: 'InferenceService',
+        plural: 'inferenceservices',
+      },
```
Comment (christianvogt) [inline frontend/src/api/k8s/__tests__/inferenceServices.spec.ts:224]:
```suggestion
      model: InferenceServiceModel,
```

**Pull request merged**